### ISSUE TITLE
[2.5.20, Joomla 4] Call to undefined method Joomla\CMS\Factory::getXML() in template.php

### ISSUE THREAD

**Issue by GHSVS-de** [templaza/astroid-framework#191]
- After migration Joomla 3.10.11 to 4.22.
- PHP 7.4
- Astroid 2.5.20
- A JoomDev template that works without issues on Joomla 3.

In frontend I get a white page with error

```
0 Call to undefined method Joomla\CMS\Factory::getXML()
/libraries/astroid/framework/template.php:60
```

It comes from these lines (method `templateVersion()`): https://github.com/templaza/astroid-framework/blob/v2.5.20/framework/template.php#L58-L63

Background: Method `JFactory::getXML()` has been removed in Joomla 4. See https://docs.joomla.org/Potential_backward_compatibility_issues_in_Joomla_4#Factory

> Factory::getXml has been removed along with JXMLElement. Use SimpleXMLElement directly instead.

I haven't made further investigations yet but the method `templateVersion()` itself has a bug with Joomla 4.

**Comment by GHSVS-de** [comment]
I could fix it like this. Tested on J3 and J4.

## Old

```
public function templateVersion()
{
	$xml = JFactory::getXML(JPATH_SITE . "/templates/{$this->template}/templateDetails.xml");
	$version = (string) $xml->version;
	return $version;
}
```

## New

```
public function templateVersion()
{
	$xml = Astroid\Helper::getXML(JPATH_SITE . "/templates/{$this->template}/templateDetails.xml");
	$version = (string) $xml->version;
	return $version;
}
```

**Comment by sonvnn** [comment]
Hello,

Thank you very much!. I fixed this issue as your suggestion on https://github.com/templaza/astroid-framework/commit/2f967f0842d7f198a1877cc610261ec1f5312c58

Thank you for your great work! If you love to contribute to Astroid, please tell me, I will add you as a Collaborator.

Best Regards,
Sonny

**Comment by GHSVS-de** [comment]
> If you love to contribute to Astroid, please tell me, I will add you as a Collaborator.

Yes, OK. Pull Requests that you can check first would be OK for me. 

----

I've found a similiar bug in `helper.php`. Same environment like above. Also an older JoomDev template on Joomla 4.

`0 Call to undefined method Joomla\CMS\Factory::getXML() /libraries/astroid/framework/helper.php:905`

Method `public static function frameworkVersion()`.

Changing

### OLD
`$xml = JFactory::getXML(JPATH_ADMINISTRATOR . '/manifests/libraries/astroid.xml');`

to

### NEW
`$xml = Astroid\Helper::getXML(JPATH_ADMINISTRATOR . '/manifests/libraries/astroid.xml');`

worked for me.
